Adds pnpm as new package manager

SerekKiri opened the pull request:
package.json --> update build script
pnpfile.js --> check for jquery.i18n
shrinkwrap.yaml --> all depedencies from pnpm

Closes https://github.com/coala/gci-leaders/issues/168

andrewda commented on .travis.yml:
```diff
@@ -2,15 +2,24 @@ language: python
+  - npm i pnpm -g
+  - pnpm install
+  - cd node_modules
+  - cd jquery.i18n
```
The previous two lines should be shortened to

```
- cd node_modules/jquery.i18n
```

andrewda commented on .travis.yml:
```diff
@@ -2,15 +2,24 @@ language: python
+  - git clone --recurse-submodules https://github.com/santhoshtr/CLDRPluralRuleParser.git
+  - cd ..
+  - cd ..
+  - cd ..
```
This can be shortened to

```
- cd ../../..
```

andrewda commented on .travis.yml:
```diff
@@ -2,15 +2,24 @@ language: python
+  - cd jquery.i18n
+  - mkdir libs
+  - cd libs
+  - git clone --recurse-submodules https://github.com/santhoshtr/CLDRPluralRuleParser.git
```
coala is angry that this line is too long

jayvdb commented on package.json:
```diff
@@ -27,13 +27,15 @@
     "mustache": "^2.3.0",
     "netlify-env": "^0.0.4",
     "node-fetch": "^1.7.3",
+    "pnpm": "2.17.7",
```
this isnt needed, as it is optional --- just like npm isnt listed here , because it isnt a dependency -- no code in this package depends on code from npm or pnpm.  They are just tools.

jayvdb replied:
not removed yet

jayvdb commented on .travis.yml:
```diff
@@ -2,15 +2,24 @@ language: python
+  - pnpm install
+  - cd node_modules
+  - cd jquery.i18n
+  - mkdir libs
```
you can do

```yaml
- mkdir -p node_modules/jquery.i18n/libs`
- cd node_modules/jquery.i18n/libs
- git clone ...
- cd -
- pnpm install
```

jayvdb replied:
not done yet

jayvdb replied:
decided to not do this

SerekKiri committed:
Adds pnpm as new package manager

package.json --> update build script
pnpfile.js --> check for jquery.i18n
shrinkwrap.yaml --> all depedencies from pnpm

Closes https://github.com/coala/gci-leaders/issues/168

jayvdb commented on .travis.yml:
```diff
@@ -1,16 +1,24 @@
 language: python
 python: 3.6
 
+
```
remove extra blank line

jayvdb commented on package.json:
```diff
@@ -27,13 +27,15 @@
     "find-rss": "^1.6.4",
     "glob": "^7.1.2",
     "graphql-client": "^2.0.0",
+    "iconv-lite": "^0.4.24",
```
this should be moved into pnpmfile.js 

which dependency has a dependency on iconv-lite ?

jayvdb commented on pnpmfile.js:
```diff
@@ -0,0 +1,14 @@
+function readPackage(pkg) {
+  switch (pkg.name) {
+    case 'webpack':
+      pkg.dependencies['jquery-i18n'] = 'github:wikimedia/jquery.i18n'
```
can you recheck this is necessary ?

jayvdb reviewed (changes requested):
needs a tidy up

SerekKiri committed:
app.js: Fixes lint errors

src/js/app.js --> Fixed lint errors

jayvdb commented:
ack https://github.com/coala/gci-leaders/pull/170/commits/565304128ed5ff8a79cf4dc7c328525dd3b58f5f https://github.com/coala/gci-leaders/pull/170/commits/0d06cfbeebc9f9a80e1c705437be04ee0c2d576b

jayvdb commented:
@gitmate-bot ff

jayvdb commented:
@gitmate-bot ff
